docs/running-on-mesos.md

```diff
@@ -501,23 +503,78 @@ See the [configuration page](configuration.html) for information on Spark config
     spark.mesos.driver.secret.names or spark.mesos.driver.secret.values will be
     written to the provided file. Paths are relative to the container's work
     directory.  Absolute paths must already exist.  Consult the Mesos Secret
-    protobuf for more information.
+    protobuf for more information. Note: File-based secrets require a custom
+    <a href="http://mesos.apache.org/documentation/latest/secrets/">SecretResolver
+    module</a>. Example:
+    
+    <pre>filename1,filename2</pre>
   </td>
 </tr>
 <tr>
   <td><code>spark.mesos.driver.secret.names</code></td>
   <td><code>(none)</code></td>
   <td>
     A comma-separated list of secret references. Consult the Mesos Secret
-    protobuf for more information.
+    protobuf for more information. Example:
+    
+    <pre>secretname1,secretname2</pre>
   </td>
 </tr>
 <tr>
   <td><code>spark.mesos.driver.secret.values</code></td>
   <td><code>(none)</code></td>
   <td>
     A comma-separated list of secret values. Consult the Mesos Secret
-    protobuf for more information.
+    protobuf for more information. Example:
+    
+    <pre>secretvalue1,secretvalue2</pre>
+  </td>
+</tr>
+
+<tr>
+  <td><code>spark.mesos.executor.secret.envkeys</code></td>
+  <td><code>(none)</code></td>
+  <td>
+    A comma-separated list that, if set, the contents of the secret referenced
+    by spark.mesos.executor.secret.names or spark.mesos.executor.secret.values will be
+    set to the provided environment variable in the executor's process. Example:
+    
+    <pre>ENVKEY1,ENVKEY2</pre>
+  </td>
+  </tr>
+  <tr>
+<td><code>spark.mesos.executor.secret.filenames</code></td>
+  <td><code>(none)</code></td>
+  <td>
+    A comma-separated list that, if set, the contents of the secret referenced by
```

Comment:
This is another description that is very weirdly worded. What I can understand of this:

"A comma-separated list of file names to which secret names and values will be written."

But I can't really know whether that's the case because it's really confusing.